sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -152,19 +179,27 @@ object JDBCRDD extends Logging {
       requiredColumns: Array[String],
       filters: Array[Filter],
       parts: Array[Partition],
-      options: JDBCOptions): RDD[InternalRow] = {
+      options: JDBCOptions,
+      outputSchema: Option[StructType] = None,
+      groupByColumns: Option[Array[FieldReference]] = None): RDD[InternalRow] = {
```

Comment:
shall we pass `Array[FieldReference]` and use empty array to represent no grouping?

Reply:
I guess no. I am using this `groupByColumns` to tell if aggregate is pushed down. Having this as an Option, I can do this:
```
if (groupByColumns.isEmpty) {  // no pushed down aggregate
  quote columns
} else {
  // columns already quoated
  if (groupByColumns.get.isEmpty) { // no group by columns
    getGroupByClause = ""
  }  else {
    getGroupByClause = ...
  }
}

```